frontend/templates/product/sections/ReviewsSection/index.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+   Box,
+   Button,
+   Flex,
+   Grid,
+   GridItem,
+   Heading,
+   HStack,
+   Icon,
+   Link,
+   Progress,
+   Stack,
+   Tag,
+   Text,
+} from '@chakra-ui/react';
+import { Rating, RatingStar, RatingStarAppearance } from '@components/ui';
+import { useAppContext } from '@ifixit/app';
+import { PageContentWrapper } from '@ifixit/ui';
+import type { Product } from '@models/product';
+import { ProductVariant } from '@models/product';
+import React from 'react';
+import { FaShieldAlt } from 'react-icons/fa';
+import { useProductReviews } from '../../hooks/useProductReviews';
+
+const INITIAL_VISIBILE_REVIEWS = 3;
+
+export type ReviewsSectionProps = {
+   product: Product;
+   selectedVariant: ProductVariant;
+};
+
+export function ReviewsSection({
+   product,
+   selectedVariant,
+}: ReviewsSectionProps) {
+   const appContext = useAppContext();
+   const reviewsQuery = useProductReviews(product);
+   const [visibleReviewsCount, setVisibleReviewsCount] = React.useState(
+      INITIAL_VISIBILE_REVIEWS
+   );
+
+   const reviewsData = reviewsQuery.data;
+
+   const totalReviewsCount = reviewsData?.count ?? 0;
+
+   const reviewCountsByRating = React.useMemo(() => {
+      if (reviewsData?.groupedReviews == null) {
+         return [];
+      }
+      const reviewsCount = Object.keys(reviewsData.groupedReviews).reduce(
+         (list, current) => {
+            const rating = parseInt(current, 10);
+            const count = reviewsData?.groupedReviews?.[current] ?? 0;
+            list.push({ rating, count });
```

Comment:
Minor: I'm curious why we're using this data structure. The name `reviewCountsByRating` would make me think it would be `Record<int, int>`, where the key is the rating and the value is the count, which would prevent duplicates of the same rating (although I doubt we'd have that problem in practice).

Did you choose this structure because `reviewCountsByRating.map(({rating, count}) => ...)` is clearer than `Object.keys(reviewCountsByRating).map(rating => ...)`?

Reply:
I try intuitively to answer since @dhmacs is on vacation this week.

Even though we have to pay a bit of extra computation for building `reviewCountsByRating`, thanks to the useMemo if `reviewsData?.groupedReviews` is stable we will do it only once. 
On the other hand this allows us to avoid executing `Object.keys` and extracting the number of reviews on every render.

Finally, it is a bit more readable since destructuring inline allows a cleaner markup.